stix_shifter_modules/rhacs/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,506 @@
+
+    @staticmethod
+    def _parse_mapped_fields(value, comparator, mapped_fields_array):
+        """
+        parse mapped fields into query expression
+        :param value: str
+        :param comparator: str
+        :param mapped_fields_array: list
+        :return: str
+        """
+        comparison_string = ""
+        for mapped_field in mapped_fields_array:
+            comparison_string += f'{mapped_field}{comparator}{value}'
+
+        return comparison_string
+
+    def _lookup_comparison_operator(self, expression_operator):
+        """
+        lookup operators support in rhacs connector
+        :param expression_operator:enum object
+        :return str
+        """
+        if str(expression_operator) not in self.comparator_lookup:
+            raise NotImplementedError(
+                f'Comparison operator {expression_operator.name} unsupported for rhacs connector')
+
+        return self.comparator_lookup[str(expression_operator)]
+
+    def _eval_comparison_value(self, expression, mapped_field_type, mapped_fields_array):
+        """
+        Function for parsing comparison expression value
+        :param expression: expression object
+        :param mapped_field_type:str
+        :param mapped_fields_array:list
+        :return: formatted expression value
+        """
+        if expression.comparator == ComparisonComparators.Matches:
+            value = QueryStringPatternTranslator._format_match(expression.value, mapped_field_type)
+
+        elif expression.comparator in [ComparisonComparators.GreaterThan,
+                                       ComparisonComparators.GreaterThanOrEqual,
+                                       ComparisonComparators.LessThan,
+                                       ComparisonComparators.LessThanOrEqual,
+                                       ComparisonComparators.Equal,
+                                       ComparisonComparators.NotEqual]:
+            value = self._format_value_type(expression.value, mapped_field_type, mapped_fields_array)
+            self._check_value_comparator_support(expression.comparator, mapped_field_type)
+            if mapped_fields_array[0] == "Violation Time":
+                value = f'{value}'
+            else:
+                value = self._format_equality(value)
+
+        elif expression.comparator == ComparisonComparators.Like:
+            value = self._format_value_type(expression.value, mapped_field_type, mapped_fields_array)
+            value = self._format_like(value, mapped_field_type)
+        else:
+            raise NotImplementedError('Unknown comparator expression operator')
+        return value
+
+    def _eval_combined_comparison_exp(self, expression):
+        """
+        Function for parsing combined comparison expression
+        :param expression: expression object
+        """
+        operator = self._lookup_comparison_operator(expression.operator)
+        if operator == '+':
+            operator = '%2B'
```

Comment:
you should encode characters in python way not hardcode it.

Reply:
Updated as per suggestion.